armory/scenarios/ucf101.py

```diff
@@ -0,0 +1,197 @@
+                        acc = float(y[0] == y_test)
+                        acc_top5 = float(y_test in y)
+                        accuracies.update(acc, 1)
+                        accuracies_top5.update(acc_top5, 1)
+                logger.info(
+                    f"Top-1 video accuracy = {accuracies.avg}, "
+                    f"top-5 video accuracy = {accuracies_top5.avg}"
+                )
+
+        # Evaluate ART classifier on test examples
+        logger.info("Running inference on benign test examples...")
+        logger.info(f"Loading testing dataset {config['dataset']['name']}...")
+        classifier.set_learning_phase(False)
+        test_data_generator = load_dataset(
+            config["dataset"],
+            epochs=1,
+            split_type="test",
+            preprocessing_fn=preprocessing_fn,
+        )
+
+        test_accuracies = AverageMeter()
+        test_accuracies_top5 = AverageMeter()
+        video_count = 0
+        for x_tests, y_tests in tqdm(test_data_generator, desc="Benign"):
+            for x_test, y_test in zip(x_tests, y_tests):
+                y = classifier.predict(x_test)
+                y = np.argsort(np.mean(y, axis=0))[-5:][::-1]
+                acc = float(y[0] == y_test)
+                acc_top5 = float(y_test in y)
+
+                test_accuracies.update(acc, 1)
+                test_accuracies_top5.update(acc_top5, 1)
+
+                logger.info(
+                    "\t ".join(
+                        [
+                            f"Video[{video_count}] : ",
+                            f"top5 = {y}",
+                            f"top1 = {y[0]}",
+                            f"true = {y_test}",
+                            f"top1_video_acc = {test_accuracies.avg}",
+                            f"top5_video_acc = {test_accuracies_top5.avg}",
+                        ]
+                    )
+                )
+                video_count += 1
+
+        logger.info(
+            "Top-1 test video accuracy = {}, top-5 test video accuracy = {}".format(
+                test_accuracies.avg, test_accuracies_top5.avg
+            )
+        )
+
+        # Evaluate the ART classifier on adversarial test examples
+        logger.info("Generating / testing adversarial examples...")
+
+        # Generate adversarial test examples
+        attack_config = config["attack"]
+        attack_module = importlib.import_module(attack_config["module"])
+        attack_fn = getattr(attack_module, attack_config["name"])
+
+        classifier.set_learning_phase(False)
+        test_data_generator = load_dataset(
+            config["dataset"],
+            epochs=1,
+            split_type="test",
+            preprocessing_fn=preprocessing_fn,
+        )
+
+        adv_accuracies = AverageMeter()
+        adv_accuracies_top5 = AverageMeter()
+        video_count = 0
+        for _ in tqdm(range(len(test_data_generator) // 10), desc="Attack"):
```

Comment:
Why `//10`? Just to make it quicker?

Reply:
It was from the original. My guess is yes, to make it quicker.

Reply:
I think that we should have a (configurable) way to subsample without hard coding into the scenario. Thoughts?